agent/connect/ca/provider_vault_test.go

```diff
@@ -200,6 +197,7 @@ func TestVaultCAProvider_SignLeaf(t *testing.T) {
 
 				// Make sure we can validate the cert as expected.
 				require.NoError(connect.ValidateLeaf(rootPEM, cert, []string{intPEM}))
+				require.Equal('\n', rune(cert[len(cert)-1]))
```

Comment:
It might be helpful to turn this terse assertion into a named test function that the different providers could share. Possibly something named like `requireTrailingNewline(t, cert)`